sos/plugins/kdump.py

```diff
@@ -27,13 +27,71 @@ class RedHatKDump(KDump, RedHatPlugin):
     files = ('/etc/kdump.conf',)
     packages = ('kexec-tools',)
 
+    def kdump_parse_fs(self):
+        device = None
+        for line in self.kdumpconf:
+            if line.startswith(("ext2", "ext3", "ext4", "nfs", "xfs")):
+                try:
+                    device = line.split()[1]
+                except (IndexError):
+                    return False
+        if device:
+            try:
+                fp = open('/etc/fstab', 'r')
+                lines = fp.readlines()
+                fp.close()
+            except IOError:
+                return False
+
+            for line in lines:
+                if line.startswith((device)):
+                    try:
+                        self.fs = line.split()[1].rstrip('/')
+                        return True
+                    except (IndexError):
+                        return False
+        return False
+
+    def kdump_parse_path(self):
+
+        for line in self.kdumpconf:
+            if line.startswith("path"):
+                try:
+                    self.path = line.split()[1]
+                    return True
+                except (IndexError):
+                    return False
+
+        return False
+
+    def read_kdump_conffile(self):
+        try:
+            fp = open('/etc/kdump.conf', 'r')
+            self.kdumpconf = fp.readlines()
+            fp.close()
+            return True
+
+        except IOError:
+            return False
+
     def setup(self):
         self.add_copy_spec([
             "/etc/kdump.conf",
-            "/etc/udev/rules.d/*kexec.rules",
-            "/var/crash/*/vmcore-dmesg.txt"
+            "/etc/udev/rules.d/*kexec.rules"
         ])
 
+        if self.read_kdump_conffile():
+            if not self.kdump_parse_fs():
+                self.fs = ""
+            if not self.kdump_parse_path():
+                self.path = "/var/crash"
+        else:
```

Comment:
Using member variables to pass data between methods is confusing - having the default set dozens of lines away from the decision on the value means you have to scroll up and down all the time.

A better style is to just return the values from the functions, assign them once (have the function *return* the default, if that's the case to use), and then you can put all the exception handling into this block and not have it spread out through all the methods, along with the booleans.